source/dub/compilers/dmd.d

```diff
@@ -156,23 +156,23 @@ class DmdCompiler : Compiler {
 
 	void setTarget(ref BuildSettings settings, in BuildPlatform platform, string tpath = null) const
 	{
-		final switch (settings.targetType) {
-			case TargetType.autodetect: assert(false, "Invalid target type: autodetect");
-			case TargetType.none: assert(false, "Invalid target type: none");
-			case TargetType.sourceLibrary: assert(false, "Invalid target type: sourceLibrary");
-			case TargetType.executable: break;
-			case TargetType.library:
-			case TargetType.staticLibrary:
-				settings.addDFlags("-lib");
-				break;
-			case TargetType.dynamicLibrary:
-				version (Windows) settings.addDFlags("-shared");
-				else settings.addDFlags("-shared", "-fPIC");
-				break;
-		}
-
-		if (tpath is null)
+		if (tpath is null) {
```

Comment:
The function is still being used to set the output file.  The only difference is that if tpath is set to something, then it means we're outputting an object file instead of the final target type.  In that case, we want to ignore the switch statement since that will also set the output type to something else (like a static library) and create a bunch of files that look like object files but aren't really.